### PR TITLE
chore: Migrate to pnpm and modularise Nx tasks

### PR DESCRIPTION
**Pull request by lachlancollins**
Changes:
- `pnpm import` to clone existing locked versions
- Setup pnpm workspaces
- Remove `yarn.lock` and lerna
- Split tsc and vitest into per-package scripts rather than root-level scripts (can be parallelised by Nx)
- Fix broken react dependency resolution
- Replace `rm -rf` (unix-specific) with `rimraf`

CI testing time:
- Before: 42s
- After: 31s

### CODE REVIEW

**Comment by lachlancollins** [comment]
@KevinVandy I've made some great progress here! I reckon I'll be able to submit the tsup PR after this.

**Review by KevinVandy — APPROVED** [review]
I've tried to find anything broken with this, but don't see anything. Thanks for the hard work. We'll see if publishing still works

File: packages/vue-table/package.json [hunk 1]
```diff
@@ -42,8 +42,12 @@
     "build/umd/*",
     "src"
   ],
+  "scripts": {
+    "clean": "rimraf ./build",
+    "test:types": "tsc"
+  },
   "dependencies": {
-    "@tanstack/table-core": "8.11.2"
+    "@tanstack/table-core": "workspace:*"
```
Comment (KevinVandy) [inline packages/vue-table/package.json:50]: Doesn't this need to tied to a version?

Comment (KevinVandy) [reply packages/vue-table/package.json:50]: This gets installed in a user's node_modules

Comment (lachlancollins) [reply packages/vue-table/package.json:50]: Pnpm will handle replacing workspace fields with versions during publishing - you can see more about it here: https://pnpm.io/workspaces#publishing-workspace-packages. And whatever files are in node_modules would be symlinks back to the local folder :)

**Pull request merged**